src/net_processing.cpp

```diff
@@ -387,6 +394,16 @@ struct CNodeState {
     //! Whether this peer relays txs via wtxid
     bool m_wtxid_relay{false};
 
+    /** Number of addr messages that can be processed from this peer. Start at 1 to
+     *  permit self-announcement. */
+    double m_addr_token_bucket{1.0};
+    /** When m_addr_token_bucket was last updated */
+    std::chrono::microseconds m_addr_token_timestamp{GetTime<std::chrono::microseconds>()};
+    /** Total number of addresses that were dropped due to rate limiting. */
+    std::atomic<uint64_t> m_addr_rate_limited{0};
+    /** Total number of addresses that were processed (excludes rate limited ones). */
+    std::atomic<uint64_t> m_addr_processed{0};
```

Comment:
These don't strictly need to be atomic since they're guarded by cs_main now, but there's no harm in leaving them like this.

Reply:
Agree, I may address this if I retouch.

Reply:
Yeah, no need to change the branch. Just pointing it out in case it confused other reviewers.